x/deployment/types/types_test.go

```diff
@@ -0,0 +1,53 @@
+package types_test
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/ovrclk/akash/x/deployment/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type GStateTest struct {
+	state       types.GroupState
+	expStateStr string
+}
+
+func (g *GStateTest) Test() func(t *testing.T) {
+	return func(t *testing.T) {
+		assert.Equal(t, types.GroupStateToReadable(g.state), g.expStateStr)
+	}
+}
+
+func TestGroupState(t *testing.T) {
+	tests := []GStateTest{
+		{
+			state:       types.GroupOpen,
+			expStateStr: "open",
+		},
+		{
+			state:       types.GroupOrdered,
+			expStateStr: "ordered",
+		},
+		{
+			state:       types.GroupMatched,
+			expStateStr: "matched",
+		},
+		{
+			state:       types.GroupInsufficientFunds,
+			expStateStr: "insufficient",
+		},
+		{
+			state:       types.GroupClosed,
+			expStateStr: "closed",
+		},
+		{
+			state:       types.GroupState(99),
+			expStateStr: "UNKNOWN",
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(strconv.Itoa(i), test.Test())
```

Comment:
this is super duper unnecessary.  try to not use a bunch of machinery in tests when it can be done in a simpler way.

```go
tests := []struct{
  state types.GroupState
  val   string
} {
  { types.GroupMatched, "matched" },
  { types.GroupOrdered, "ordered" },
  // ...
}

for _, test := range tests {
  assert.Equal(t,t.val,types.GroupStateToReadable(t.state))
}
```

Reply:
Alright, I added additional assertions to the `gStateTest struct Test()func(t*testing.T)` method, to work with our linter and t.Run. If you want all of the assertions to be with in the `for { }` block, though, then t.Run chokes on passage of the `test` variable. 

So I removed all of that and left it as a single testing.T that iterates and logs initial conditions. That's the only way to have more than one assertion, and not violate the `scopelinter` + `t.Run()` without returning a test function from the struct AFAIK.

Reply:
Thanks.  Yeah, the `t.Run(name,test.generate())` thing is an overkill here.